### ISSUE TITLE
Allow re-auth

### ISSUE THREAD

**Issue by lwansbrough** [naia-lib/naia#10]
Hey there, I've got a suggestion for auth: It would be helpful to be able to re-auth at any point, in the case where the user's session is going to expire while the user is connected.

**Comment by naia-rs** [comment]
Yeah definitely! Thanks for the suggestion. There's a couple ways to do this, I don't have an opinion around the implementation at the moment.

My initial thought is that NaiaServer could expose a method like: send_reauth_request(key: UserKey, timeout: Duration), which triggers a ReauthRequest event on NaiaClient, which must be responded to using a send_reauth_response(reauthEvent: Event).
Then in NaiaServer you could either reuse the "on_auth()" callback method, or have a new "on_reauth()" callback, that processes the reauth event payload.
If a user does not successfully re-auth within the given timeout, then they are forcibly disconnected.

Anyone else have thoughts on this?

**Comment by connorcarpenter** [comment]
Status check on this one, with a concrete read of what it costs against the handshake as it stands today.

**Where re-auth would have to live.** Today auth is a *pre-connection* HTTP exchange: the client POSTs credentials, the server answers 200 with an identity token (which it then expects in the in-band handshake) or 401. Once the datachannel is up, that path is gone — there is no auth channel left open, and `MainUser::take_auth_address` deliberately consumes the auth address so it can only be answered once. So re-auth cannot reuse the auth transport; it has to be a new message exchange over the established connection.

**What the sketch above translates into.** Following your four steps:

1. `send_reauth_request(key, timeout)` — server-side; needs a per-user deadline tracked the same way `max_pending_auth_users` now tracks pending-auth age, and a sweep that disconnects on expiry.
2. `ReauthRequest` client event — a new handshake-adjacent header, or an internal message on a reserved reliable channel. The latter is cheaper and reuses the existing delivery guarantees.
3. `send_reauth_response(event)` → a server-side `ReauthEvent` carrying the payload, decoded against `message_kinds` exactly like the initial `AuthEvent`. Reusing `on_auth` would be ambiguous (the app cannot tell a first auth from a re-auth, and the two want different failure handling), so a separate event is better.
4. Timeout → forcible disconnect, which the server already has in `user_delete`.

**The part that needs a decision before anyone writes code:** what happens to in-flight state during re-auth. Does replication keep running while the answer is outstanding (simple, but means an expired session keeps receiving world updates for up to `timeout`), or does the user's scope freeze until they re-auth (safer, more moving parts)? That choice, not the plumbing, is the design question here.

Leaving open. It is a real feature with a clear shape; it is not blocked on anything, just unstarted.

**Comment by connorcarpenter** [comment]
Fixed on `dev` (f37d6676): `Server::disconnect_user_with` now attaches a reason message to a server-initiated kick, and the client reports `DisconnectReason::Kicked` instead of `ClientDisconnected`.
